_meta/config/common.p2.yml.tmpl

```diff
@@ -9,6 +9,9 @@ outputs:
     # username: "elastic"
     # password: "changeme"
 
+# Here you can configure your list of inputs. You can either configure all the inputs as a list of arrays
+# or create an "inputs.d" directory and put a yaml file inside for each input. If you use the inputs.d
+# directory, it is important that each .yml file repeats the `inputs:` top level array part.
```

Comment:
```suggestion
# Here you can configure your list of inputs. You can either configure all the inputs as a list of arrays
# or create an "inputs.d" directory containing your input configurations. See $link for how to structure the "inputs.d" directory.
```

I think it would be better to link directly to a section of the official documentation that describes how `inputs.d` works with examples instead of trying to describe it completely here.

`inputs.d/` apparently doesn't have a dedicated heading or section in our documentation, making it hard to find and know about. It is only mentioned briefly as part of the several paragraphs introducing the standalone agent configuration: https://www.elastic.co/guide/en/fleet/current/elastic-agent-configuration.html. The source for this is [here](https://github.com/elastic/ingest-docs/blob/dbda6bde9e28e370da8319f5edc795cb874d3404/docs/en/ingest-management/elastic-agent/configuration/elastic-agent-configuration.asciidoc?plain=1#L24-L29)

Reply:
That is the link I was looking for, search failed me. Should we make it a separate paragraph to make it easier to find?

The more fundamental question: If we make it more visible, we kind of make it also a recommended way to configure elastic-agent. My take: If you have many inputs configured, it makes it nicely manageable.

Reply:
Updated the config with your change and the link.

Reply:
> Should we make it a separate paragraph to make it easier to find?
I would make inputs.d a sub-section/heading so we can link to it directly.

> The more fundamental question: If we make it more visible, we kind of make it also a recommended way to configure elastic-agent. My take: If you have many inputs configured, it makes it nicely manageable.

My view is we present both options and let users decide which one they prefer. We already support inputs.d with no plans to remove it, that would be a breaking change. The only reason I wouldn't encourage it would be if we were planning to deprecate it.